File: StudentBookingV2/app/src/main/java/com/danieleroccaforte/ium/studentbooking/MainActivity.java
package com.danieleroccaforte.ium.studentbooking;

import android.annotation.SuppressLint;
import android.content.Intent;
import android.os.Bundle;
import android.view.View;

import androidx.appcompat.app.AppCompatActivity;
import androidx.fragment.app.Fragment;

import com.danieleroccaforte.ium.studentbooking.fragments.AccountFragment;
import com.danieleroccaforte.ium.studentbooking.fragments.CoursesFragment;
import com.danieleroccaforte.ium.studentbooking.fragments.SearchFragment;
import com.google.android.material.bottomnavigation.BottomNavigationView;

public class MainActivity extends AppCompatActivity implements View.OnClickListener{
    BottomNavigationView bottomNavigationView;
    @SuppressLint("NonConstantResourceId")
    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        getSupportFragmentManager().beginTransaction().replace(R.id.fragment_container, new AccountFragment()).commit();

        bottomNavigationView = findViewById(R.id.bottom_navigation);
        bottomNavigationView.setOnNavigationItemSelectedListener(item -> {
            Fragment fragment = null;
            switch (item.getItemId()){
                case R.id.my_account:
                    fragment = new AccountFragment();
                    break;
                case R.id.my_courses:
                    fragment = new CoursesFragment();
                    break;
                case R.id.search:
                    fragment = new SearchFragment();
                    break;
            }
            assert fragment != null;
            getSupportFragmentManager().beginTransaction().replace(R.id.fragment_container, fragment).commit();
            return true;
        });
    }

    @SuppressLint("NonConstantResourceId")
    @Override
    public void onClick(View view) {
        if (view.getId() == R.id.logout_button) {
            Intent intent = new Intent(getApplicationContext(), Login.class);
            startActivity(intent);
            finish();
        }
    }
}

File: StudentBooking V1/app/src/main/java/com/danieleroccaforte/ium/studentbooking/MainActivity.kt
@file:Suppress("DEPRECATION")

package com.danieleroccaforte.ium.studentbooking

import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.provider.ContactsContract
import androidx.fragment.app.Fragment
import androidx.lifecycle.MediatorLiveData
import androidx.lifecycle.MutableLiveData
import kotlinx.android.synthetic.main.dashboard_layout.*

class MainActivity : AppCompatActivity() {

    private val AccountFragment = com.danieleroccaforte.ium.studentbooking.fragments.AccountFragment()
    private val CercaFragment = com.danieleroccaforte.ium.studentbooking.fragments.CercaFragment()
    private val CorsiFragment = com.danieleroccaforte.ium.studentbooking.fragments.CorsiFragment()

    private val emailLiveData = MutableLiveData<String>()
    private val passwordLiveData = MutableLiveData<String>()
    private val isValidLiveData = MediatorLiveData<Boolean>().apply {
        addSource(emailLiveData){ email ->
            val password = passwordLiveData.value
            this.value = validateForm(email, password)
        }
    }

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.fragment_login)



        replaceFragment(AccountFragment)

        //LISTENER PER CAMBIO SCHERMATA
        bottom_navigation.setOnNavigationItemSelectedListener {
            when(it.itemId){
                R.id.ic_account -> replaceFragment(AccountFragment)
                R.id.ic_mycourses -> replaceFragment(CorsiFragment)
                R.id.ic_search -> replaceFragment(CercaFragment)
            }
            true
        }
    }

    private fun validateForm(email: String?, password: String?) : Boolean{
        
        return false
    }

    private fun replaceFragment(fragment: Fragment){
        if(fragment != null){
            val transaction = supportFragmentManager.beginTransaction()
            transaction.replace(R.id.fragment_container, fragment)
            transaction.commit()
        }
    }
}